src/utils/FileEmitter.ts

```diff
@@ -367,6 +404,49 @@ export class FileEmitter {
 		return this.assignReferenceId(consumedChunk, emittedChunk.id);
 	}
 
+	private emitPrebuiltChunk(
+		emitPrebuiltChunk: Omit<EmittedFile, 'fileName' | 'name'> &
+			Pick<EmittedPrebuiltChunk, 'exports' | 'map'>
+	): string {
+		if (typeof emitPrebuiltChunk.code !== 'string') {
+			return error(
+				errorFailedValidation(
+					`Emitted prebuilt chunks need to have a valid string code, received "${emitPrebuiltChunk.code}"`
+				)
+			);
+		}
+		if (
+			typeof emitPrebuiltChunk.fileName !== 'string' ||
+			!(
+				typeof emitPrebuiltChunk.fileName === 'string' &&
+				!isPathFragment(emitPrebuiltChunk.fileName)
+			)
+		) {
+			return error(
+				errorFailedValidation(
+					`Emitted prebuilt chunks need to have a valid string fileName and that is neither absolute nor relative paths, received "${emitPrebuiltChunk.fileName}"`
+				)
+			);
+		}
+		const consumedPrebuiltChunk: ConsumedPrebuiltChunk = {
+			code: emitPrebuiltChunk.code,
+			exports: emitPrebuiltChunk.exports,
+			fileName: emitPrebuiltChunk.fileName,
+			map: emitPrebuiltChunk.map,
+			referenceId: '',
+			type: 'prebuilt-chunk'
+		};
+		const referenceId = this.assignReferenceId(
+			consumedPrebuiltChunk,
+			consumedPrebuiltChunk.fileName
+		);
+		if (this.output) {
+			this.output.bundle[consumedPrebuiltChunk.fileName] =
```

Comment:
This scenario is where a plugin emits a prebuilt-chunk during output generation. To test this, e.g. emit such a chunk in `generateBundle`.

Reply:
done